### PR TITLE
Upgrade kube go libraries

### PR DESCRIPTION
**Pull request by ivanovaleksandar**
The outdated client-go libraries that were used are messing with the change introduced here: https://github.com/estafette/estafette-extension-gke/pull/83 the API in version `0.17.0` did not have the contract for `.spec.ingressClassName`, so after a ingress is deployed, the background running CloudFlare DNS extension is actually changing the ingress the a version of an ingress resources with `0.17.0` type of contract.

This is simply fixed with updating the libraries and adapt the methods used.

### CODE REVIEW

File: main.go [hunk 1]
```diff
@@ -83,6 +84,9 @@ func main() {
 	// init log format from envvar ESTAFETTE_LOG_FORMAT
 	foundation.InitLoggingFromEnv(foundation.NewApplicationInfo(appgroup, app, version, branch, revision, buildDate))
 
+	// create context to cancel commands on sigterm
+	ctx := foundation.InitCancellationContext(context.Background())
```
Comment (snigdhasambitak) [inline main.go]: Why is this required for the cloudflare dns? This is suppose to be running as a controller in the background. Do we need to cancel the contexts on terminations? sigterm triggers the cancel function and for a controller which is a long running object, do we want to include this

Comment (ivanovaleksandar) [reply main.go]: I think we can do it either way with or without canceling context, but this is case it cannot reach the kube API for some reason. It gives a safeguard timeouts. What do you think? Keep in mind that this is only for the kube API comms, not CloudFlare DNS.

Comment (snigdhasambitak) [reply main.go]: We are already doing graceful shutdown on the application level : https://github.com/estafette/estafette-cloudflare-dns/blob/main/main.go#L180. So for the kube-api we are hadling this in a way. Don’t see any harm in adding this. But was curious if sigterms were not handled before

Comment (snigdhasambitak) [reply main.go]: @ivanovaleksandar I think its fine to add this. Please test the changes in playground to catch any unexpected behaviours

**Review by msharbaji — APPROVED** [review]

File: main.go [hunk 2]
```diff
@@ -605,7 +614,12 @@ func makeIngressChanges(cf *Cloudflare, kubeClientset *kubernetes.Clientset, ing
 			ingress.Annotations[annotationCloudflareState] = string(cloudflareStateByteArray)
 
 			// update ingress, because the state annotations have changed
-			_, err = kubeClientset.NetworkingV1beta1().Ingresses(ingress.Namespace).Update(ingress)
+			_, err = kubeClientset.NetworkingV1().Ingresses(ingress.Namespace).Update(ctx, ingress, metav1.UpdateOptions{
+				TypeMeta:        metav1.TypeMeta{},
+				DryRun:          []string{},
```
Comment (snigdhasambitak) [inline main.go]: Do we need to explicitly define DryRun? and do we want modifications should not be persisted?

File: main.go [hunk 3]
```diff
@@ -384,7 +388,12 @@ func makeServiceChanges(cf *Cloudflare, kubeClientset *kubernetes.Clientset, ser
 		service.Annotations[annotationCloudflareState] = string(cloudflareStateByteArray)
 
 		// update service, because the state annotations have changed
-		service, err = kubeClientset.CoreV1().Services(service.Namespace).Update(service)
+		service, err = kubeClientset.CoreV1().Services(service.Namespace).Update(ctx, service, metav1.UpdateOptions{
+			TypeMeta:        metav1.TypeMeta{},
+			DryRun:          []string{},
+			FieldManager:    "",
```
Comment (snigdhasambitak) [inline main.go]: If we are not specifying the actor then adding FieldManager explicitely doesn’t bring any value

File: main.go [hunk 4]
```diff
@@ -384,7 +388,12 @@ func makeServiceChanges(cf *Cloudflare, kubeClientset *kubernetes.Clientset, ser
...
+			FieldValidation: "",
```
Comment (snigdhasambitak) [inline main.go]: I think the default value has to be `Ignore`, `warn` or `strict`. If its default then probably not defining this is a good idea

File: main.go (hunk 2 above)
Comment (msharbaji) [reply main.go]: Yes, maybe you can just pass empty object metav1.UpdateOptions{}

File: main.go [hunk 5]
```diff
@@ -384,7 +388,12 @@ func makeServiceChanges(cf *Cloudflare, kubeClientset *kubernetes.Clientset, ser
 		service.Annotations[annotationCloudflareState] = string(cloudflareStateByteArray)
 
 		// update service, because the state annotations have changed
-		service, err = kubeClientset.CoreV1().Services(service.Namespace).Update(service)
+		service, err = kubeClientset.CoreV1().Services(service.Namespace).Update(ctx, service, metav1.UpdateOptions{
```
Comment (snigdhasambitak) [inline main.go]: if using default options then can just have it blank `metav1.UpdateOptions{}`

Comment (ivanovaleksandar) [reply main.go]: You are right. The vim pluggin auto expanded those on save. Reverting those to the empty structs.

File: main.go (hunk 2 above)
Comment (ivanovaleksandar) [reply main.go]: Removed.

File: main.go (hunk 3 above)
Comment (ivanovaleksandar) [reply main.go]: Removed.

File: main.go (hunk 4 above)
Comment (ivanovaleksandar) [reply main.go]: Removed.

**Review by snigdhasambitak — APPROVED** [review]

File: main.go (hunk 1 above)
Comment (msharbaji) [inline main.go]: @ivanovaleksandar here you don't need to add initCancellationContext as we are passing foundation.InitGracefulShutdownHandling() in the line 118. just you can create the context with 
ctx : = context.Background()

**Review by msharbaji — CHANGES_REQUESTED** [review]

File: main.go (hunk 1 above)
Comment (snigdhasambitak) [reply main.go]: @msharbaji that was my understanding as well;)

Comment (ivanovaleksandar) [reply main.go]: We discussed this with @snigdhasambitak little bit already above, but I can create it as just a normal context. Doesn't have to be one with cancellation.

Comment (ivanovaleksandar) [reply main.go]: ok, created normal context instead now.

**Review by msharbaji — APPROVED** [review]
